Support STM with LL library, support ASCII chars directly, added much more glyphs

Vinzenz82 opened the pull request:
Hello Coders,

I added following features:

- support of LL library for STM32 target
- support of directly using ASCII chars - enables to use sprintf output directly
- support more glyphs : latin alphabet as described here: https://en.wikipedia.org/wiki/Seven-segment_display

`sprintf(cData, "HELLO_-~");`
![Hello](https://github.com/MahdaSystem/TM1638/assets/125079210/dd586ea8-ed00-4e20-a14c-81ebc7715d62)

`sprintf(cData, "_-~coder");`
![coder](https://github.com/MahdaSystem/TM1638/assets/125079210/35565ea6-98f7-481d-b25d-276b8fb67169)

`sprintf(cData, "_-~go~-_");`
![go](https://github.com/MahdaSystem/TM1638/assets/125079210/2e9d44bc-c6fa-43fd-9a95-62829e9c4393)

`sprintf(cData, "~-_on_-~");`
![on](https://github.com/MahdaSystem/TM1638/assets/125079210/1fd99fbd-c621-4c87-85b1-442b040eab41)

`sprintf(cData, "start at");`
![start_at](https://github.com/MahdaSystem/TM1638/assets/125079210/a5d3d539-8b6f-48ac-957d-3fa534b19b4f)

`sprintf(cData, "%2d-%2d-%2d", Now.hour, Now.minutes, Now.seconds);`
![clock](https://github.com/MahdaSystem/TM1638/assets/125079210/20e787da-e543-4b8c-96fe-a6942d872c9f)

Tested with different controllers.

With best regards from germany,

Vinzenz Petr

Hossein-M98 commented:
Hello Vinzenz
Thanks for your contribution.
I will check your changes and if everything is OK I will merge them into the main code.
Respectfully, 
Hossein from Iran :)

Hossein-M98 reviewed (changes requested):
There are some problems that should be fixed before merging into the master branch:
1. In the platform files, create a new platform for the STM32 LL library, instead of changing the previous STM32 HAL-compatible platform.
2. Delete this part of the code. Because there are no hex numbers marked "g".
![image](https://github.com/MahdaSystem/TM1638/assets/52778819/fb8cd899-b09f-468c-a103-c8d904a02cfc)

Good luck

Vinzenz82 commented:
Dear Hossein,

thank you very much for fast feedback and response. I removed the test code (0x40). In the platform code I already added a new target, look for TM1638_PLATFORM_STM32_LL. The original TM1638_PLATFORM_STM32 is still working with HAL and is not touched.

Best Regards,

Vinzenz
